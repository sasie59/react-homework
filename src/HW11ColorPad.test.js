import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './HW11ColorPad';

test('should HW11ColorPad', () => {
  const { container } = render(<HW />);
  expect(container).toMatchSnapshot();
});

test('should change rgb range', () => {
  const { container, getByTestId} = render(<HW />);
  const redInput = getByTestId('red');
  const greenInput = getByTestId('green');
  const blueInput = getByTestId('blue');
  
  userEvent.type(redInput, '200');
  userEvent.type(greenInput, '10');
  userEvent.type(blueInput, '173');

  expect(container).toMatchSnapshot();
});

