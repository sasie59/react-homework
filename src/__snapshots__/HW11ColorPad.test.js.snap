// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should HW11ColorPad 1`] = `
<div>
  <div>
    <h1>
      HW11ColorPad
    </h1>
    <h1>
      調色盤
    </h1>
    <div
      style="width: 200px; height: 200px; margin: 0px auto; background-color: rgb(0, 150, 150);"
    />
    R
    <input
      data-testid="red"
      max="255"
      min="0"
      type="range"
    />
    0
    <br />
    G
    <input
      data-testid="green"
      max="255"
      min="0"
      type="range"
    />
    150
    <br />
    B
    <input
      data-testid="blue"
      max="255"
      min="0"
      type="range"
    />
    150
    <br />
  </div>
</div>
`;

exports[`should change rgb range 1`] = `
<div>
  <div>
    <h1>
      HW11ColorPad
    </h1>
    <h1>
      調色盤
    </h1>
    <div
      style="width: 200px; height: 200px; margin: 0px auto; background-color: rgb(200, 10, 173);"
    />
    R
    <input
      data-testid="red"
      max="255"
      min="0"
      type="range"
    />
    200
    <br />
    G
    <input
      data-testid="green"
      max="255"
      min="0"
      type="range"
    />
    10
    <br />
    B
    <input
      data-testid="blue"
      max="255"
      min="0"
      type="range"
    />
    173
    <br />
  </div>
</div>
`;
